eclipse/tests/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/staticanalysis/MarkupModelListenerTest.java

```diff
@@ -438,9 +438,28 @@ private void runForActivePage(Consumer<IWorkbenchPage> consumer) {
         });
     }
 
+	private List<StaticAnalysisWarningEvent> process_marker_and_return_generated_warning_list(Runnable runnable) throws Exception {
+		List<StaticAnalysisWarningEvent> list = new ArrayList<>();
+
+		Mockito.doAnswer(new Answer<Object>() {
+
+			@SuppressWarnings("unchecked")
+			@Override
+			public Object answer(InvocationOnMock invocation) throws Throwable {
+				Stream<StaticAnalysisWarningEvent> stream = (Stream<StaticAnalysisWarningEvent>) invocation.getArguments()[0];
+				list.addAll(stream.collect(Collectors.toList()));
+				return null;
+			}}).when(this.trackingEventManager).addEvents(Mockito.any());
+
+		runnable.run();
+
+		this.saveWorkspaceAndWaitForBuild();
+
+		return list;
+	}
 
     private List<StaticAnalysisWarningEvent> deleteMarkerAndReturnGeneratedWarningList(IMarker marker) throws Exception {
```

Comment:
Wouldn't it make more sense to name deleteMarkerAndReturnGeneratedWarningList "delete_marker_and_return_generated_warning_list", in the same style as you named process_marker_and_return_generated_warning_list()?

Reply:
Actually, I had tried to keep `processMarkerAndReturnGeneratedWarningList` as is, but due to a CheckStyle bug it kept marking this method as public. My suspicion was because of the public method `answer` in the Mockito anonymous class. I was unable to reproduce this issue yet in a fresh project.

Since I could not address this issue, I am okay with renaming this method as well, although my ideal case would be that the private methods keep their camelCase signature.